apps/server/src/adapters/WebsocketAdapter.ts

```diff
@@ -49,8 +49,22 @@ export class SocketServer implements IAdapter {
 
     this.wss.on('connection', (ws) => {
       let clientId = getRandomName();
-      this.clientIds.add(clientId);
-      logger.info(LogOrigin.Client, `${this.clientIds.size} Connections with new: ${clientId}`);
+      this.clients.set(clientId, { type: ClientTypes.Unknown, identify: false, redirect: '' });
+      logger.info(LogOrigin.Client, `${this.clients.size} Connections with new: ${clientId}`);
+
+      // give the client a change to tell us thire prefed name
+      const giveNameTimeout = setTimeout(
+        () =>
+          ws.send(
+            JSON.stringify({
+              type: 'client-name',
+              payload: clientId,
+            }),
+          ),
+        100,
+      );
+      //then send the client list
+      setTimeout(() => this.sendClientList(), 120);
```

Comment:
I think adding side effects this way could give us a chance for hard to follow bugs.
I wonder if it is best for us to live with the renaming happening straight after connection